aea/cli/publish.py

```diff
@@ -47,6 +49,20 @@ def publish(ctx: Context, registry):
         publish_agent(ctx)
 
 
+def _check_is_item_in_local_registry(
+    public_id, item_type_plural, registry_path
+):
+    try:
+        try_get_item_source_path(
+            registry_path,
+            public_id.author,
+            item_type_plural,
+            public_id.name
+        )
+    except click.ClickException as e:
+        raise click.ClickException('Dependency is missing. {}'.format(e))
```

Comment:
Maybe provide more info: `Dependency {} is missing. Please push it first and then retry.`

Reply:
What is missing is described in inner exception that we catch and re-raise with additional info. The rest will add now.